### PR TITLE
Define NSLog macro as mentioned in the TF docs

### LINKED ISSUE #3427: Fix iOS logs in TestFairy
### Description

*Type*: Bug

*Summary*: TestFairy logs stopped working on iOS

#### Expected behavior
Logs are present

#### Actual behavior

In order to see logs, please update your project's Prefix Header (.pch):
#import "TestFairy.h"
#define NSLog(s, ...) do { NSLog(s, ##__VA_ARGS__); TFLog(s, ##__VA_ARGS__); } while (0)

### Reproduction
[comment]: # (Describe how we can replicate the bug step by step.)

- Open Status on iOS
- go to TestFairy session

### Additional Information
* Status version: at least latest PRs
* Operating System: iOS

### PR DESCRIPTION
**Pull request by pedropombeiro**
[comment]: # (Please replace ... with your information. Remove < and >)
[comment]: # (To auto-close issue on merge, please insert the related issue number after # i.e fixes #566)

fixes #3427

### Summary:

[comment]: # (Summarise the problem and how the pull request solves it)
This PR adds the `#define` mentioned by @asemiankevich in #3427 and found in the TestFairy documentation.

status: ready

### CODE REVIEW

File: modules/react-native-status/ios/RCTStatus/RCTStatus.h [hunk 1]
```diff
@@ -4,6 +4,8 @@
 #import <JavaScriptCore/JavaScriptCore.h>
 #import "Jail.h"
 
+#define NSLog(s, ...) do { NSLog(s, ##__VA_ARGS__); TFLog(s, ##__VA_ARGS__); } while (0)
```
Comment (mandrigin) [inline modules/react-native-status/ios/RCTStatus/RCTStatus.h]: We need to make sure that that doesn't go to any of the app store builds though.
@rasom is the best person to talk about that :)

Comment (pedropombeiro) [reply modules/react-native-status/ios/RCTStatus/RCTStatus.h]: In app store builds, won't `TESTFAIRY_ENABLED` be set to 0 (from `.env.prod`)?

Comment (mandrigin) [reply modules/react-native-status/ios/RCTStatus/RCTStatus.h]: yeah, I was expecting something like
```objc
#if defined(TESTFAIRY_ENABLED) && TESTFAIRY_ENABLED
   #define NSLog(s, ...)....
#endif
```

Comment (pedropombeiro) [reply modules/react-native-status/ios/RCTStatus/RCTStatus.h]: I have to confess I assumed `TFLog` would be a no-op if `TESTFAIRY_ENABLED` wasn't set to `1` (this commit was meant to serve for testing first and not really PR material). Good catch!
I don't think `TESTFAIRY_ENABLED` gets turned into a preprocessor directive though (I see it being accessed with `[ReactNativeConfig envFor:@"TESTFAIRY_ENABLED"]`. I think I'll need help from a dev who's familiar with ObjC to find out the best way to do this. Hopefully `TFLog` is innocuous enough if the API token isn't set and we can just leave the `NSLog` definition as-is.

Comment (mandrigin) [reply modules/react-native-status/ios/RCTStatus/RCTStatus.h]: Even if it is, I like being explicit 🤓

**Review by rasom — APPROVED** [review]

**Review by jeluard — APPROVED** [review]

**Comment by annadanchenko** [comment]
@PombeirP no build is available for testing. last attempt to build failed with 
** ARCHIVE FAILED **


The following build commands failed:
	CompileC /Users/jenkins/Library/Developer/Xcode/DerivedData/StatusIm-hfhpdsizfexsabgxvrmzrqbnmyig/Build/Intermediates.noindex/ArchiveIntermediates/StatusIm/IntermediateBuildFilesPath/RCTStatus.build/Release-iphoneos/RCTStatus.build/Objects-normal/arm64/Jail.o Jail.m normal arm64 objective-c com.apple.compilers.llvm.clang.1_0.compiler
(1 failure) 

see https://jenkins.status.im/job/status-react/job/pull%20requests/view/change-requests/job/PR-3541/3/console

**Comment by asemiankevich** [comment]
Its not working to me yet, do we need to re-trigger newish build or something? Still no logs on ios https://app.testfairy.com/projects/4803590-status/builds/7898030/sessions/1/?accessToken=ymSOp6TqRYvwb-wrDtL2U7UoDQ4

<blockquote><div><strong><a href="https://app.testfairy.com/projects/4803590-status/builds/7898030/sessions/1/?accessToken=ymSOp6TqRYvwb-wrDtL2U7UoDQ4">TestFairy: Status 0.0.2309 (2306.3) @1</a></strong></div></blockquote>

**Comment by annadanchenko** [comment]
can confirm that no logs are shown if use build 5 for this PR. TF session without log: https://app.testfairy.com/projects/4803590-status/builds/7898030/sessions/2/?accessToken=a0sV4A/85qNQBFZ01wxkRddZyfc
Branch: PR-3541
Android: https://i.diawi.com/Dwxkh7
iOS: https://i.diawi.com/uZHmc7

<blockquote><div><strong><a href="https://app.testfairy.com/projects/4803590-status/builds/7898030/sessions/2/?accessToken=a0sV4A/85qNQBFZ01wxkRddZyfc">TestFairy: Status 0.0.2309 (2306.3) @2</a></strong></div></blockquote>
<blockquote><div><strong><a href="https://i.diawi.com/Dwxkh7">Install - Diawi - Development and In-house Apps Wireless Installation</a></strong></div></blockquote>
<blockquote><div><strong><a href="https://i.diawi.com/uZHmc7">Install - Diawi - Development and In-house Apps Wireless Installation</a></strong></div></blockquote>

**Comment by dmitryn** [comment]
@annadanchenko could you please try this build?

Branch: PR-3541
Android: https://i.diawi.com/j48KMn
iOS: https://i.diawi.com/pLmMxb


It works for me on iOS simulator, here is my TF session: https://app.testfairy.com/projects/4803590-status/builds/7452515/sessions/13

<blockquote><div><strong><a href="https://app.testfairy.com/projects/4803590-status/builds/7452515/sessions/13">TestFairy: Login</a></strong></div></blockquote>

**Comment by dmitryn** [comment]
@annadanchenko new build is ready:
Branch: PR-3541
Android: https://i.diawi.com/wzkatT
iOS: https://i.diawi.com/NxrbeR

<blockquote><div><strong><a href="https://i.diawi.com/wzkatT">Install - Diawi - Development and In-house Apps Wireless Installation</a></strong></div></blockquote>
<blockquote><div><strong><a href="https://i.diawi.com/NxrbeR">Install - Diawi - Development and In-house Apps Wireless Installation</a></strong></div></blockquote>

**Pull request merged**